crates/bevy_render/src/renderer/headless_render_resource_context.rs

```diff
@@ -17,6 +17,9 @@ use std::{ops::Range, sync::Arc};
 pub struct HeadlessRenderResourceContext {
     buffer_info: Arc<RwLock<HashMap<BufferId, BufferInfo>>>,
     texture_descriptors: Arc<RwLock<HashMap<TextureId, TextureDescriptor>>>,
+    texture_view_descriptors: Arc<RwLock<HashMap<TextureViewId, TextureViewDescriptor>>>,
+    texture_texture_views:
+        Arc<RwLock<HashMap<TextureId, HashMap<Option<BindGroupDescriptorId>, TextureViewId>>>>,
```

Comment:
Is there a reason these are using `RwLock` instead of `Mutex`. 
From the usage of them. they are _not_ public, and and only times they are used they are only ever _written_ to. 
If we don't need specific read/write access having a `Mutex` over a `RwLock` would be preferred.

Reply:
I reverse checked all of them by using "find usages" in my IDE and looked for ones which only had `write` usages. Three of them could easily be turned into `Mutex`es, but the others seem to be used in read operations quite frequently.

I'm ignoring the diagnostics system though, since that reads from nearly all of them.

Reply:
If there are valid `reads` for a field, then I think it's ok to keep the `RwLock`. 
However, if something is only every written to (which is a little weird imo) then it should definitely be `Mutex`. 

Even arguably, if the filed is **not highly contended** you could use `Mutex`, but I'm not familiar enough with the code to make that judgement.

Reply:
> However, if something is only every written to (which is a little weird imo)

We kind of only "write" to a bunch of things, but in many instances we extract something from the `HashMap`, so in a sense we're also reading it.

> Even arguably, if the filed is not highly contended you could use Mutex, but I'm not familiar enough with the code to make that judgement.

I don't know enough about all of these fields either, I just looked to see if their `.read`s were used anywhere.